aurora/integration-tests/src/lib.rs

```diff
@@ -129,20 +136,36 @@ mod tests {
                 &self.engine,
                 check_result,
             )
-            .await;
+            .await
         }
 
-        pub async fn approve_spend_mock_tokens(&self) {
+        pub async fn approve_spend_mock_tokens(&self, user_account: Option<Account>) {
             approve_spend_tokens(
                 &self.aurora_mock_token,
                 self.aurora_fast_bridge_contract.address,
-                &self.user_account,
+                &user_account.unwrap_or(self.user_account.clone()),
                 &self.engine,
             )
             .await;
         }
 
-        pub async fn init_token_transfer(&self, amount: u128, fee_amount: u128, valid_till: u64) {
+        pub async fn init_token_transfer(
+            &self,
+            amount: u128,
+            fee_amount: u128,
+            valid_till: Option<u64>,
+            user_address: Option<Address>,
+            user_account: Option<Account>,
+            check_output: bool,
+            gas: Option<u64>,
+        ) {
+            let valid_till = valid_till.unwrap_or(
+                (std::time::SystemTime::now()
+                    .duration_since(std::time::SystemTime::UNIX_EPOCH)
+                    .unwrap()
+                    .as_nanos()
+                    + Duration::from_secs(TRANSFER_EXPIRATION_PERIOD_S).as_nanos()) as u64,
```

Comment:
```suggestion
                    + Duration::from_secs(TRANSFER_EXPIRATION_PERIOD_SEC).as_nanos()) as u64,
```

Reply:
fixed: https://github.com/aurora-is-near/fast-bridge-protocol/pull/101/commits/fc520f211b971e073aae04aa32c26491e606f6d5